### PR TITLE
add the API: GET /area-types/{area-type}/parents

### PR DESCRIPTION
**Pull request by mohsdev**
### What

Describe what you have changed and why.

### How to review

Describe the steps required to test the changes.

### Who can review

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

File: contract/area_types.go [hunk 1]
```diff
@@ -11,3 +11,14 @@ type GetAreaTypesRequest struct {
 type GetAreaTypesResponse struct {
 	AreaTypes []model.AreaType `json:"area-types"`
 }
+
+// GetAreaTypeParentsRequest defines the schema for the GET /area-types/{area-type}/parents query parameter
+type GetAreaTypeParentsRequest struct {
+	Dataset  string `schema:"dataset"`
+	AreaType string `schema:"area-type"`
+}
```
Comment (PauloPortugal) [inline contract/area_types.go:19]: Why do we need this?
The enpoint is `GET /area-types/{area-type}/parents` and there are no query parameters.

Please remove this.

File: handler/area_types.go [hunk 2]
```diff
@@ -65,3 +65,48 @@ func (h *AreaTypes) Get(w http.ResponseWriter, r *http.Request) {
...
+	if err := schema.NewDecoder().Decode(&request, r.URL.Query()); err != nil {
+		h.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			errors.Wrap(err, "failed to decode query parameters"),
+		)
+		return
+	}
+
```
Comment (PauloPortugal) [inline handler/area_types.go:83]: I believe this is not required as we do not have query parameters.

File: handler/area_types.go [hunk 3]
```diff
@@ -65,3 +65,48 @@ func (h *AreaTypes) Get(w http.ResponseWriter, r *http.Request) {
...
+	cantabularResponse, err := h.ctblr.GetGeographyDimensions(ctx, request.Dataset)
```
Comment (PauloPortugal) [inline handler/area_types.go:84]: I don't understand why we are querying the `GetGeographyDimensions` endpoint.
Shouldn't we be querying the `dp-api-client-go` endpoint and create a GraphQL query?

File: handler/area_types.go [hunk 4]
```diff
@@ -65,3 +65,48 @@ func (h *AreaTypes) Get(w http.ResponseWriter, r *http.Request) {
...
+			dperrors.StatusCode(err), // Can be changed to ctblr.StatusCode(err) once added to Client
+			errors.Wrap(err, "failed to get area-types"),
+		)
+		return
+	}
+
+	var response contract.GetAreaTypeParentsResponse
+	if cantabularResponse != nil {
+		for _, edge := range cantabularResponse.Dataset.RuleBase.IsSourceOf.Edges {
+			if request.AreaType == edge.Node.Name {
```
Comment (PauloPortugal) [inline handler/area_types.go:98]: This may be inefficient, as we are getting all of the Dataset Edges and then transversing it to check if it matches the `AreaType` provided.
Also `request.AreaType` will be an empty string at this stage as we are mapping it from the query parameters which do not exist.

**Review by PauloPortugal — CHANGES_REQUESTED** [review]
Please review the comments.

Also we should also update the PR description instead of just relying on the template version.

Also you will need to write component tests to provide the minimum coverage to this new endpoint, which should include _Happy path_ and _unhappy paths_ please.

It also seems that we should not query all the dataset edges and do a match in memory as this could be very inefficient. 
We should instead get the `area-type` from the url parameter, not the query parameter and call an existing or new endpoint that should return all the parents for that `area-type`.

**Comment by andre-urbani** [comment]
Repository is being decommissioned
